diff --git a/src/Makefile.am b/src/Makefile.am
index 141209fa1b47..58bc5e627d44 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -73,7 +73,7 @@ libfyaml_la_CPPFLAGS = $(AM_CPPFLAGS) $(INTERNAL_INCLUDES)
 libfyaml_la_CFLAGS = $(AM_CFLAGS)
 libfyaml_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(AM_LIBLDFLAGS) \
-		      -version $(LIBTOOL_VERSION)
+		      -version-number $(LIBTOOL_VERSION)
 libfyaml_la_LIBADD = $(LIBM)
 
 # blake3 section
