diff --git a/src/modules/simpleclock.cpp b/src/modules/simpleclock.cpp
index b6a96eccf022..58ecbfcd0dac 100644
--- a/src/modules/simpleclock.cpp
+++ b/src/modules/simpleclock.cpp
@@ -1,3 +1,5 @@
+#define _POSIX_C_SOURCE
+
 #include "modules/simpleclock.hpp"
 
 #include <time.h>
@@ -16,8 +18,9 @@ waybar::modules::Clock::Clock(const std::string& id, const Json::Value& config)
 
 auto waybar::modules::Clock::update() -> void {
   tzset();  // Update timezone information
-  auto now = std::chrono::system_clock::now();
-  auto localtime = fmt::localtime(std::chrono::system_clock::to_time_t(now));
+  time_t now = time(NULL);
+  struct tm localtime;
+  localtime_r(&now, &localtime);
   auto text = fmt::format(fmt::runtime(format_), localtime);
   label_.set_markup(text);
 
