diff --git a/src/main.c b/src/main.c
index 087c483..a4eb799 100644
--- a/src/main.c
+++ b/src/main.c
@@ -40,9 +40,7 @@
 #include <sys/types.h>
 #endif
 
-#if !defined __MINGW32__
-#define WAVE_USE_XID
-#else
+#ifdef HAVE_USE_XID
 #undef WAVE_USE_XID
 #endif
 
diff --git a/src/twinwave.c b/src/twinwave.c
index 71ba15a..3a7f125 100644
--- a/src/twinwave.c
+++ b/src/twinwave.c
@@ -143,15 +143,6 @@ if(GDK_IS_WAYLAND_DISPLAY(gdk_display_get_default()))
 	use_embedded = 0;
 	}
 #endif
-	{
-	xsocket[0] = gtk_socket_new ();
-	xsocket[1] = gtk_socket_new ();
-	gtk_widget_show (xsocket[0]);
-	gtk_widget_show (xsocket[1]);
-	}
-
-if(!twinwayland)
-g_signal_connect(XXX_GTK_OBJECT(xsocket[0]), "plug-removed", G_CALLBACK(plug_removed), NULL);
 
 #if GTK_CHECK_VERSION(3,0,0)
 main_vbox = gtk_box_new(GTK_ORIENTATION_VERTICAL, 5);
@@ -208,11 +199,7 @@ if(hMapFile != NULL)
 				memset(&pi, 0, sizeof(PROCESS_INFORMATION));
 
 				sprintf(buf, "0+%08X", shmid);
-#ifdef MINGW_USE_XID
-				sprintf(buf2, "%x", gtk_socket_get_id (GTK_SOCKET(xsocket[0])));
-#else
 				sprintf(buf2, "%x", 0);
-#endif
 
 				arglist[0] = "gtkwave.exe";
 				arglist[1] = "-D";
@@ -279,11 +266,7 @@ if(hMapFile != NULL)
 				memset(&pi, 0, sizeof(PROCESS_INFORMATION));
 
 				sprintf(buf, "1+%08X", shmid);
-#ifdef MINGW_USE_XID
-				sprintf(buf2, "%x", gtk_socket_get_id (GTK_SOCKET(xsocket[1])));
-#else
 				sprintf(buf2, "%x", 0);
-#endif
 
 				arglist[0] = "gtkwave.exe";
 				arglist[1] = "-D";
@@ -427,15 +410,6 @@ if(shmid >=0)
 				char **arglist = calloc(n_items, sizeof(char *));
 
 				sprintf(buf, "0+%08X", shmid);
-				if(use_embedded)
-					{
-#ifdef MAC_INTEGRATION
-					sprintf(buf2, "%x", gtk_socket_get_id (GTK_SOCKET(xsocket[0])));
-#else
-					sprintf(buf2, "%lx", (long)gtk_socket_get_id (GTK_SOCKET(xsocket[0])));
-#endif
-					}
-					else
 					{
 					sprintf(buf2, "%x", 0);
 					}
@@ -465,15 +439,6 @@ if(shmid >=0)
 			char **arglist = calloc(n_items, sizeof(char *));
 
 			sprintf(buf, "1+%08X", shmid);
-			if(use_embedded)
-				{
-#ifdef MAC_INTEGRATION
-				sprintf(buf2, "%x", gtk_socket_get_id (GTK_SOCKET(xsocket[1])));
-#else
-				sprintf(buf2, "%lx", (long)gtk_socket_get_id (GTK_SOCKET(xsocket[1])));
-#endif
-				}
-				else
 				{
 				sprintf(buf2, "%x", 0);
 				}
