From 614c05be0556d6f03d40108869cc6814134089b0 Mon Sep 17 00:00:00 2001
From: Konstantin Bogdanov <konstantin@clickhouse.com>
Date: Tue, 17 Jun 2025 23:52:54 +0200
Subject: [PATCH] Remove std::char_traits usage

---
 nanodbc/nanodbc.cpp | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/nanodbc/nanodbc.cpp b/nanodbc/nanodbc.cpp
index fb85598a40f9..e76eb174d143 100644
--- a/nanodbc/nanodbc.cpp
+++ b/nanodbc/nanodbc.cpp
@@ -260,8 +260,14 @@ constexpr std::size_t size(const T (&array)[N]) noexcept
 template <std::size_t N>
 inline std::size_t size(NANODBC_SQLCHAR const (&array)[N]) noexcept
 {
-    auto const n = std::char_traits<NANODBC_SQLCHAR>::length(array);
+    std::size_t len = 0;
+    NANODBC_SQLCHAR const* s = array;
+    for (; *s != '\0'; ++s)
+        ++len;
+
+    std::size_t const n = len;
     NANODBC_ASSERT(n < N);
+
     return n < N ? n : N - 1;
 }
 
@@ -3563,9 +3569,8 @@ std::list<datasource> list_datasources()
                 "incompatible SQLCHAR and string::value_type");
 
             datasource dsn;
-            dsn.name = string(&name[0], &name[std::char_traits<NANODBC_SQLCHAR>::length(name)]);
-            dsn.driver =
-                string(&driver[0], &driver[std::char_traits<NANODBC_SQLCHAR>::length(driver)]);
+            dsn.name = string(&name[0], &name[size(name)]);
+            dsn.driver = string(&driver[0], &driver[size(driver)]);
 
             dsns.push_back(std::move(dsn));
             direction = SQL_FETCH_NEXT;
@@ -3616,7 +3621,7 @@ std::list<driver> list_drivers()
                 "incompatible SQLCHAR and string::value_type");
 
             driver drv;
-            drv.name = string(&descr[0], &descr[std::char_traits<NANODBC_SQLCHAR>::length(descr)]);
+            drv.name = string(&descr[0], &descr[size(descr)]);
 
             // Split "Key1=Value1\0Key2=Value2\0\0" into list of key-value pairs
             auto beg = &attrs[0];
-- 
2.50.1

