diff --git a/meson.build b/meson.build
index a0cf675..3d266b5 100644
--- a/meson.build
+++ b/meson.build
@@ -513,7 +513,7 @@ if get_option('tools')
   mkdir_p = 'mkdir -p "$DESTDIR@0@"'
   meson.add_install_script('sh', '-c', mkdir_p.format(sbindir))
 
-  ln_s = 'ln -sfr "$DESTDIR@0@/kmod" "$DESTDIR@1@"'
+  ln_s = 'ln -sf "$DESTDIR@0@/kmod" "$DESTDIR@1@"'
   foreach tool : _tools
     meson.add_install_script('sh', '-c', ln_s.format(bindir, sbindir / tool))
   endforeach
