diff --git a/Source/WebCore/contentextensions/ContentExtensionActions.h b/Source/WebCore/contentextensions/ContentExtensionActions.h
index a600f410bf96..2d4c3ccd1029 100644
--- a/Source/WebCore/contentextensions/ContentExtensionActions.h
+++ b/Source/WebCore/contentextensions/ContentExtensionActions.h
@@ -29,6 +29,7 @@
 
 #if ENABLE(CONTENT_EXTENSIONS)
 
+#include <system_error>
 #include <WebCore/ContentExtensionStringSerialization.h>
 #include <system_error>
 #include <wtf/JSONValues.h>
