diff --git a/compiler/rustc_target/src/spec/base/linux_musl.rs b/compiler/rustc_target/src/spec/base/linux_musl.rs
index a134a7c..a048812 100644
--- a/compiler/rustc_target/src/spec/base/linux_musl.rs
+++ b/compiler/rustc_target/src/spec/base/linux_musl.rs
@@ -4,6 +4,7 @@ pub(crate) fn opts() -> TargetOptions {
     TargetOptions {
         env: "musl".into(),
         crt_static_default:  false,
+        no_default_libraries: false,
         ..base::linux::opts()
     }
 }
