diff --git a/cmake/aom_install.cmake b/cmake/aom_install.cmake
index 1ac795e1b3e1..4f5db3092c3c 100644
--- a/cmake/aom_install.cmake
+++ b/cmake/aom_install.cmake
@@ -80,11 +80,7 @@ macro(setup_aom_install_targets)
       endif()
     endif()
 
-    if(BUILD_SHARED_LIBS)
-      set(AOM_INSTALL_LIBS aom aom_static)
-    else()
-      set(AOM_INSTALL_LIBS aom)
-    endif()
+    set(AOM_INSTALL_LIBS aom)
 
     set(AOM_GENERATED_DIR "${CMAKE_CURRENT_BINARY_DIR}/generated")
     set(AOM_VERSION_CONFIG
