diff --git a/src/OpenGl/OpenGl_GraphicDriver.cxx b/src/OpenGl/OpenGl_GraphicDriver.cxx
index a06c4490199e..bf0b941b13e7 100644
--- a/src/OpenGl/OpenGl_GraphicDriver.cxx
+++ b/src/OpenGl/OpenGl_GraphicDriver.cxx
@@ -40,15 +40,6 @@ IMPLEMENT_STANDARD_RTTIEXT(OpenGl_GraphicDriver, Graphic3d_GraphicDriver)
   #include <Aspect_NeutralWindow.hxx>
 #endif
 
-#if !defined(_WIN32) && !defined(__ANDROID__) && !defined(__QNX__)                                 \
-  && (!defined(__APPLE__) || defined(HAVE_XLIB))
-  #include <X11/Xlib.h> // XOpenDisplay()
-  #include <X11/Xutil.h>
-  #if !defined(__EMSCRIPTEN__)
-    #include <GL/glx.h>
-  #endif
-#endif
-
 #if !defined(HAVE_EGL)
   #if defined(__ANDROID__) || defined(__QNX__) || defined(__EMSCRIPTEN__) || defined(HAVE_GLES2)   \
     || defined(OCCT_UWP)
