diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7eba244..94dad59 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -153,6 +153,7 @@ if(NOT OPTION_DEDICATED)
             endif()
             find_package(Harfbuzz)
             find_package(ICU OPTIONAL_COMPONENTS i18n)
+            find_package(ICU OPTIONAL_COMPONENTS uc)
         endif()
     endif()
 endif()
@@ -330,6 +331,7 @@ if(NOT OPTION_DEDICATED)
     link_package(Fontconfig TARGET Fontconfig::Fontconfig)
     link_package(Harfbuzz TARGET harfbuzz::harfbuzz)
     link_package(ICU_i18n)
+    link_package(ICU_uc)
 
     if(SDL2_FOUND AND OPENGL_FOUND AND UNIX)
         # SDL2 dynamically loads OpenGL if needed, so do not link to OpenGL when
