diff --git a/js/src/jit/MacroAssembler.h b/js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h
+++ b/js/src/jit/MacroAssembler.h
@@ -1513,14 +1513,24 @@
                       Label* label) PER_SHARED_ARCH;
 
   inline void branch16(Condition cond, const Address& lhs, Imm32 rhs,
                        Label* label) PER_SHARED_ARCH;
 
-  inline void branch32(Condition cond, Register lhs, Register rhs,
-                       Label* label) PER_SHARED_ARCH;
-  inline void branch32(Condition cond, Register lhs, Imm32 rhs,
-                       Label* label) PER_SHARED_ARCH;
+  // On some platforms, it is possible to do a 32-bit comparison against
+  // the low 32 bits of a 64-bit register, ignoring the high bits. On
+  // other architectures (eg RISC-V), this may not be possible. Passing
+  // LhsHighBitsAreClean::No implies that the architecture-specific code
+  // must zero/sign-extend the low bits of the Lhs if it can't ignore
+  // the high bits.
+  enum class LhsHighBitsAreClean { Yes, No };
+
+  inline void branch32(Condition cond, Register lhs, Register rhs, Label* label,
+                       LhsHighBitsAreClean clean = LhsHighBitsAreClean::Yes)
+      PER_SHARED_ARCH;
+  inline void branch32(Condition cond, Register lhs, Imm32 rhs, Label* label,
+                       LhsHighBitsAreClean clean = LhsHighBitsAreClean::Yes)
+      PER_SHARED_ARCH;
 
   inline void branch32(Condition cond, Register lhs, const Address& rhs,
                        Label* label) DEFINED_ON(arm64);
 
   inline void branch32(Condition cond, const Address& lhs, Register rhs,
diff --git a/js/src/jit/arm/MacroAssembler-arm-inl.h b/js/src/jit/arm/MacroAssembler-arm-inl.h
--- a/js/src/jit/arm/MacroAssembler-arm-inl.h
+++ b/js/src/jit/arm/MacroAssembler-arm-inl.h
@@ -1425,17 +1425,17 @@
   ma_cmp(scratch, imm, scratch2);
   ma_b(label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Register rhs,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean) {
   ma_cmp(lhs, rhs);
   ma_b(label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Imm32 rhs,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean) {
   ScratchRegisterScope scratch(*this);
 
   ma_cmp(lhs, rhs, scratch);
   ma_b(label, cond);
 }
diff --git a/js/src/jit/arm64/MacroAssembler-arm64-inl.h b/js/src/jit/arm64/MacroAssembler-arm64-inl.h
--- a/js/src/jit/arm64/MacroAssembler-arm64-inl.h
+++ b/js/src/jit/arm64/MacroAssembler-arm64-inl.h
@@ -1166,17 +1166,17 @@
       MOZ_CRASH("unexpected condition");
   }
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Register rhs,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean) {
   cmp32(lhs, rhs);
   B(label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Imm32 imm,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean) {
   if (imm.value == 0 && cond == Assembler::Equal) {
     Cbz(ARMRegister(lhs, 32), label);
   } else if (imm.value == 0 && cond == Assembler::NotEqual) {
     Cbnz(ARMRegister(lhs, 32), label);
   } else {
diff --git a/js/src/jit/loong64/MacroAssembler-loong64-inl.h b/js/src/jit/loong64/MacroAssembler-loong64-inl.h
--- a/js/src/jit/loong64/MacroAssembler-loong64-inl.h
+++ b/js/src/jit/loong64/MacroAssembler-loong64-inl.h
@@ -1088,16 +1088,18 @@
       MOZ_CRASH("unexpected condition");
   }
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Register rhs,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean clean) {
+  // TODO: Handle LhsHighBitsAreClean::No.
   ma_b(lhs, rhs, label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Imm32 imm,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean clean) {
+  // TODO: Handle LhsHighBitsAreClean::No.
   ma_b(lhs, imm, label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, const Address& lhs, Register rhs,
                               Label* label) {
diff --git a/js/src/jit/mips-shared/MacroAssembler-mips-shared-inl.h b/js/src/jit/mips-shared/MacroAssembler-mips-shared-inl.h
--- a/js/src/jit/mips-shared/MacroAssembler-mips-shared-inl.h
+++ b/js/src/jit/mips-shared/MacroAssembler-mips-shared-inl.h
@@ -590,16 +590,18 @@
       MOZ_CRASH("unexpected condition");
   }
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Register rhs,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean clean) {
+  // TODO: Handle LhsHighBitsAreClean::No.
   ma_b(lhs, rhs, label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Imm32 imm,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean clean) {
+  // TODO: Handle LhsHighBitsAreClean::No.
   ma_b(lhs, imm, label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, const Address& lhs, Register rhs,
                               Label* label) {
diff --git a/js/src/jit/riscv64/MacroAssembler-riscv64-inl.h b/js/src/jit/riscv64/MacroAssembler-riscv64-inl.h
--- a/js/src/jit/riscv64/MacroAssembler-riscv64-inl.h
+++ b/js/src/jit/riscv64/MacroAssembler-riscv64-inl.h
@@ -504,16 +504,18 @@
     default:
       MOZ_CRASH("unexpected condition");
   }
 }
 void MacroAssembler::branch32(Condition cond, Register lhs, Register rhs,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean clean) {
+  // TODO: Handle LhsHighBitsAreClean::No.
   ma_b(lhs, rhs, label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Imm32 imm,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean clean) {
+  // TODO: Handle LhsHighBitsAreClean::No.
   ma_b(lhs, imm, label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, const Address& lhs, Register rhs,
                               Label* label) {
diff --git a/js/src/jit/wasm32/MacroAssembler-wasm32-inl.h b/js/src/jit/wasm32/MacroAssembler-wasm32-inl.h
--- a/js/src/jit/wasm32/MacroAssembler-wasm32-inl.h
+++ b/js/src/jit/wasm32/MacroAssembler-wasm32-inl.h
@@ -733,16 +733,16 @@
                               Label* label) {
   MOZ_CRASH();
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Register rhs,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean) {
   MOZ_CRASH();
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Imm32 rhs,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean) {
   MOZ_CRASH();
 }
 
 void MacroAssembler::branch32(Condition cond, wasm::SymbolicAddress lhs,
                               Imm32 rhs, Label* label) {
diff --git a/js/src/jit/x86-shared/MacroAssembler-x86-shared-inl.h b/js/src/jit/x86-shared/MacroAssembler-x86-shared-inl.h
--- a/js/src/jit/x86-shared/MacroAssembler-x86-shared-inl.h
+++ b/js/src/jit/x86-shared/MacroAssembler-x86-shared-inl.h
@@ -549,17 +549,17 @@
   cmp16(lhs, rhs);
   j(cond, label);
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Register rhs,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean) {
   cmp32(lhs, rhs);
   j(cond, label);
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Imm32 rhs,
-                              Label* label) {
+                              Label* label, LhsHighBitsAreClean) {
   cmp32(lhs, rhs);
   j(cond, label);
 }
 
 void MacroAssembler::branch32(Condition cond, const Address& lhs, Register rhs,

