diff --git a/js/src/jit/BaselineCodeGen.cpp b/js/src/jit/BaselineCodeGen.cpp
--- a/js/src/jit/BaselineCodeGen.cpp
+++ b/js/src/jit/BaselineCodeGen.cpp
@@ -2949,11 +2949,12 @@
       int32_t constantVal = data.toInt32();
 
       Label maybeDouble;
       masm.branchTestInt32(Assembler::NotEqual, value, &maybeDouble);
       masm.branch32(JSOpToCondition(op, true), value.payloadOrValueReg(),
-                    Imm32(constantVal), &pass);
+                    Imm32(constantVal), &pass,
+                    MacroAssembler::LhsHighBitsAreClean::No);
       masm.jump(&fail);
 
       masm.bind(&maybeDouble);
       {
         FloatRegister unboxedValue = FloatReg0;
@@ -3054,11 +3055,11 @@
   masm.bind(&isInt32);
   {
     Label maybeDouble;
     masm.branchTestInt32(Assembler::NotEqual, value, &maybeDouble);
     masm.branch32(JSOpToCondition(op, true), value.payloadOrValueReg(), payload,
-                  &pass);
+                  &pass, MacroAssembler::LhsHighBitsAreClean::No);
     masm.jump(&fail);
 
     masm.bind(&maybeDouble);
     {
       FloatRegister unboxedValue = FloatReg0;
diff --git a/js/src/jit/CodeGenerator.cpp b/js/src/jit/CodeGenerator.cpp
--- a/js/src/jit/CodeGenerator.cpp
+++ b/js/src/jit/CodeGenerator.cpp
@@ -1610,11 +1610,12 @@
   Register output = ToRegister(comp->output());
 
   Label fail, pass, done, maybeDouble;
   masm.branchTestInt32(Assembler::NotEqual, value, &maybeDouble);
   masm.branch32(JSOpToCondition(op, true), value.payloadOrValueReg(),
-                Imm32(constantVal), &pass);
+                Imm32(constantVal), &pass,
+                MacroAssembler::LhsHighBitsAreClean::No);
   masm.jump(&fail);
 
   masm.bind(&maybeDouble);
   {
     FloatRegister unboxedValue = ToFloatRegister(comp->temp0());
diff --git a/js/src/jit/loong64/MacroAssembler-loong64-inl.h b/js/src/jit/loong64/MacroAssembler-loong64-inl.h
--- a/js/src/jit/loong64/MacroAssembler-loong64-inl.h
+++ b/js/src/jit/loong64/MacroAssembler-loong64-inl.h
@@ -1089,17 +1089,27 @@
   }
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Register rhs,
                               Label* label, LhsHighBitsAreClean clean) {
-  // TODO: Handle LhsHighBitsAreClean::No.
+  if (clean == LhsHighBitsAreClean::No) {
+    ScratchRegisterScope scratch(asMasm());
+    as_slli_w(scratch, lhs, 0);
+    ma_b(scratch, rhs, label, cond);
+    return;
+  }
   ma_b(lhs, rhs, label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Imm32 imm,
                               Label* label, LhsHighBitsAreClean clean) {
-  // TODO: Handle LhsHighBitsAreClean::No.
+  if (clean == LhsHighBitsAreClean::No) {
+    SecondScratchRegisterScope scratch(asMasm());
+    as_slli_w(scratch, lhs, 0);
+    ma_b(scratch, imm, label, cond);
+    return;
+  }
   ma_b(lhs, imm, label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, const Address& lhs, Register rhs,
                               Label* label) {
diff --git a/js/src/jit/mips-shared/MacroAssembler-mips-shared-inl.h b/js/src/jit/mips-shared/MacroAssembler-mips-shared-inl.h
--- a/js/src/jit/mips-shared/MacroAssembler-mips-shared-inl.h
+++ b/js/src/jit/mips-shared/MacroAssembler-mips-shared-inl.h
@@ -591,17 +591,27 @@
   }
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Register rhs,
                               Label* label, LhsHighBitsAreClean clean) {
-  // TODO: Handle LhsHighBitsAreClean::No.
+  if (clean == LhsHighBitsAreClean::No) {
+    ScratchRegisterScope scratch(asMasm());
+    as_sll(scratch, lhs, 0);
+    ma_b(scratch, rhs, label, cond);
+    return;
+  }
   ma_b(lhs, rhs, label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Imm32 imm,
                               Label* label, LhsHighBitsAreClean clean) {
-  // TODO: Handle LhsHighBitsAreClean::No.
+  if (clean == LhsHighBitsAreClean::No) {
+    SecondScratchRegisterScope scratch(asMasm());
+    as_sll(scratch, lhs, 0);
+    ma_b(scratch, imm, label, cond);
+    return;
+  }
   ma_b(lhs, imm, label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, const Address& lhs, Register rhs,
                               Label* label) {
diff --git a/js/src/jit/riscv64/MacroAssembler-riscv64-inl.h b/js/src/jit/riscv64/MacroAssembler-riscv64-inl.h
--- a/js/src/jit/riscv64/MacroAssembler-riscv64-inl.h
+++ b/js/src/jit/riscv64/MacroAssembler-riscv64-inl.h
@@ -505,17 +505,29 @@
       MOZ_CRASH("unexpected condition");
   }
 }
 void MacroAssembler::branch32(Condition cond, Register lhs, Register rhs,
                               Label* label, LhsHighBitsAreClean clean) {
-  // TODO: Handle LhsHighBitsAreClean::No.
+  if (clean == LhsHighBitsAreClean::No) {
+    UseScratchRegisterScope temps(this);
+    Register scratch = temps.Acquire();
+    slliw(scratch, lhs, 0);
+    ma_b(scratch, rhs, label, cond);
+    return;
+  }
   ma_b(lhs, rhs, label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, Register lhs, Imm32 imm,
                               Label* label, LhsHighBitsAreClean clean) {
-  // TODO: Handle LhsHighBitsAreClean::No.
+  if (clean == LhsHighBitsAreClean::No) {
+    UseScratchRegisterScope temps(this);
+    Register scratch = temps.Acquire();
+    slliw(scratch, lhs, 0);
+    ma_b(scratch, imm, label, cond);
+    return;
+  }
   ma_b(lhs, imm, label, cond);
 }
 
 void MacroAssembler::branch32(Condition cond, const Address& lhs, Register rhs,
                               Label* label) {

