From 9bd483aa6643f4a57f400a2b305916adc5d136e8 Mon Sep 17 00:00:00 2001
From: Yukari Chiba <i@0x7f.cc>
Date: Tue, 28 Nov 2023 16:24:46 +0800
Subject: [PATCH] lib/init.sh: fix parsing kernel cmdline

---
 lib/init.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/init.sh b/lib/init.sh
index dd5b90a..d6a3aaa 100755
--- a/lib/init.sh
+++ b/lib/init.sh
@@ -49,7 +49,7 @@ parse_cmdline()
         # Maintain backward compatibility with kernel parameters.
         ro | rw)      rorw=$_param ;;
         rootwait)     root_wait=-1 ;;
-        --)           init_args=${_param##*--}; break ;;
+        --)           init_args=${cmdline##*--}; break ;;
         rootfstype=*) root_type=${_param#*=} ;;
         rootflags=*)  root_opts=${_param#*=} ;;
         rootdelay=*)  root_wait=${_param#*=} ;;
