diff --git a/cupsfilters/pdftoraster.cxx b/cupsfilters/pdftoraster.cxx
index 3b50030..89d5537 100644
--- a/cupsfilters/pdftoraster.cxx
+++ b/cupsfilters/pdftoraster.cxx
@@ -2193,39 +2193,3 @@ cfFilterPDFToRaster(int inputfd,            // I - File descriptor input stream
   return (ret);
 #endif // HAVE_POPPLER
 }
-#ifdef HAVE_POPPLER
-
-// Replace memory allocation methods for memory check
-// For compatibility with g++ >= 4.7 compilers _GLIBCXX_THROW
-// should be used as a guard, otherwise use traditional definition
-#ifndef _GLIBCXX_THROW
-#  if __cplusplus >= 201103L
-#    define _GLIBCXX_THROW(_EXC)
-#  else
-#    define _GLIBCXX_THROW(_EXC) throw(_EXC)
-#  endif
-#endif
-
-void * operator new(size_t size) _GLIBCXX_THROW (std::bad_alloc)
-{
-  return malloc(size);
-}
-
-void operator delete(void *p) throw ()
-{
-  free(p);
-}
-
-void * operator new[](size_t size) _GLIBCXX_THROW (std::bad_alloc)
-{
-  return malloc(size);
-}
-
-#if 0
-void operator delete[](void *p) throw ()
-{
-  free(p);
-}
-#endif // 0
-
-#endif // HAVE_POPPLER
