diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 351b03f..f9085f6 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -40,6 +40,8 @@ target_link_libraries(yaml-cpp-tests
     yaml-cpp
     gmock)
 
+target_compile_options(yaml-cpp-tests PRIVATE -Wno-c++11-narrowing)
+
 set_property(TARGET yaml-cpp-tests PROPERTY CXX_STANDARD_REQUIRED ON)
 if (NOT DEFINED CMAKE_CXX_STANDARD)
   set_target_properties(yaml-cpp-tests PROPERTIES CXX_STANDARD 11)
