diff --git a/src/igr-tests/igr-runner.cc b/src/igr-tests/igr-runner.cc
index c70d79a..1cda905 100644
--- a/src/igr-tests/igr-runner.cc
+++ b/src/igr-tests/igr-runner.cc
@@ -12,7 +12,7 @@ extern char **environ;
 
 int main(int argc, char **argv)
 {
-    const char * const test_dirs[] = { "basic", "environ", "ps-environ", "chain-to", "force-stop",
+    const char * const test_dirs[] = { "basic", "ps-environ", "chain-to", "force-stop",
             "restart", "check-basic", "check-cycle", "check-lint", "reload1", "reload2", "no-command-error",
             "add-rm-dep", "var-subst", "svc-start-fail" };
     constexpr int num_tests = sizeof(test_dirs) / sizeof(test_dirs[0]);


