diff --git a/src/vim/syntax/PKGBUILD.vim b/src/vim/syntax/PKGBUILD.vim
index 37ce89e..61de376 100644
--- a/src/vim/syntax/PKGBUILD.vim
+++ b/src/vim/syntax/PKGBUILD.vim
@@ -82,7 +82,7 @@ syn region pbBackupGroup start=/^backup=(/ end=/)/ contains=pb_k_backup,pbValidB
 
 " arch
 syn keyword pb_k_arch arch contained
-syn keyword pbArch i686 x86_64 x86_64_v3 ppc pentium4 armv7h aarch64 loong64 any contained
+syn keyword pbArch x86_64 aarch64 riscv64 loong64 any contained
 syn match pbIllegalArch /[^='"() ]/ contained contains=pbArch
 syn region pbArchGroup start=/^arch=(/ end=/)/ contains=pb_k_arch,pbArch,pbIllegalArch,pbComment
 
