diff --git a/src/scripts/lib/device-info-cache.lua b/src/scripts/lib/device-info-cache.lua
index 6e4cf10c..35fc3f86 100644
--- a/src/scripts/lib/device-info-cache.lua
+++ b/src/scripts/lib/device-info-cache.lua
@@ -33,8 +33,8 @@ function module.get_device_info (self, device)
   if not dev_info then
     local device_name = device_properties ["device.name"]
     if not device_name then
-      Log.critical (device, "invalid device.name")
-      return nil
+      Log.warning (device, "invalid device.name")
+      --return nil
     end
 
     Log.trace (device, string.format (
