diff --git a/src/btop_tools.cpp b/src/btop_tools.cpp
index 6e8becb..98fb5c6 100644
--- a/src/btop_tools.cpp
+++ b/src/btop_tools.cpp
@@ -26,6 +26,7 @@ tab-size = 4
 #include <ranges>
 #include <robin_hood.h>
 #include <widechar_width.hpp>
+#include <codecvt>
 
 #include <unistd.h>
 #include <termios.h>
@@ -283,7 +284,7 @@ namespace Tools {
 
 	auto ssplit(const string& str, const char& delim) -> vector<string> {
 		vector<string> out;
-		for (const auto& s : str 	| rng::views::split(delim)
+		for (const auto& s : str 	| rng::views::lazy_split(delim)
 									| rng::views::transform([](auto &&rng) {
 										return string_view(&*rng.begin(), rng::distance(rng));
 		})) {
