diff --git a/xfconf/abicheck.sh b/xfconf/abicheck.sh
index 7731f9f..91ec6c2 100755
--- a/xfconf/abicheck.sh
+++ b/xfconf/abicheck.sh
@@ -19,6 +19,6 @@
 #
 
 trap 'rm expected-abi actual-abi' EXIT
-${CPP:-cpp} -DINCLUDE_INTERNAL_SYMBOLS -DINCLUDE_VARIABLES -DALL_FILES ${srcdir:-.}/xfconf.symbols | sed 's/ G_GNUC.*$//;s/ PRIVATE//;/^ *$/d;/^#/d' | sort >expected-abi
-${NM:-nm} .libs/libxfconf-0.so* | awk '/ [DRTG] / {print $3}' | sort | uniq > actual-abi
+clang -E -xc -DINCLUDE_INTERNAL_SYMBOLS -DINCLUDE_VARIABLES -DALL_FILES ${srcdir:-.}/xfconf.symbols | sed 's/ G_GNUC.*$//;s/ PRIVATE//;/^ *$/d;/^#/d' | sort >expected-abi
+${NM:-nm} .libs/libxfconf-0.so | awk '/ [DRTG] / {print $3}' | sort | uniq > actual-abi
 diff -u expected-abi actual-abi
