diff --git a/libxfce4ui/abicheck.sh.in b/libxfce4ui/abicheck.sh.in
index 0d9a5a0..c1950e2 100644
--- a/libxfce4ui/abicheck.sh.in
+++ b/libxfce4ui/abicheck.sh.in
@@ -19,6 +19,6 @@
 #
 
 trap 'rm expected-abi actual-abi' EXIT
-${CPP:-cpp} -DINCLUDE_INTERNAL_SYMBOLS -DINCLUDE_VARIABLES -DALL_FILES @-DENABLE_X11@ @-DENABLE_WAYLAND@ ${srcdir:-.}/libxfce4ui.symbols | sed 's/ G_GNUC.*$//;s/ PRIVATE//;/^ *$/d;/^#/d' | sort >expected-abi
-${NM:-nm} .libs/libxfce4ui-2.so* | awk '/ [DRTG] / {print $3}' | sort | uniq > actual-abi
+clang -E -xc -DINCLUDE_INTERNAL_SYMBOLS -DINCLUDE_VARIABLES -DALL_FILES @-DENABLE_X11@ @-DENABLE_WAYLAND@ ${srcdir:-.}/libxfce4ui.symbols | sed 's/ G_GNUC.*$//;s/ PRIVATE//;/^ *$/d;/^#/d' | sort >expected-abi
+${NM:-nm} .libs/libxfce4ui-2.so | awk '/ [DRTG] / {print $3}' | sort | uniq > actual-abi
 diff -u expected-abi actual-abi
