diff --git a/C/hashes/hash.h b/C/hashes/hash.h
index 3abf275..5084b25 100644
--- a/C/hashes/hash.h
+++ b/C/hashes/hash.h
@@ -42,17 +42,7 @@
 
 #include "../7zTypes.h"
 
-#ifndef _UINT32_T_DECLARED
-typedef UInt32 uint32_t;
-#define _UINT32_T_DECLARED
-#endif
-
-#ifndef _UINT64_T_DECLARED
-typedef UInt64 uint64_t;
-#define _UINT64_T_DECLARED
-#endif
-
-//#include <stdint.h>
+#include <stdint.h>
 
 #ifndef min
 #define min(a,b) (((a)>(b))?(b):(a))
