diff --color -urN pacman-6.0.1/scripts/libmakepkg/lint_package/dotfiles.sh.in pacman-6.0.1-new/scripts/libmakepkg/lint_package/dotfiles.sh.in
--- pacman-6.0.1/scripts/libmakepkg/lint_package/dotfiles.sh.in	2021-09-04 17:42:20.357641200 +0800
+++ pacman-6.0.1-new/scripts/libmakepkg/lint_package/dotfiles.sh.in	2023-01-09 02:54:33.538888302 +0800
@@ -29,7 +29,7 @@
 
 check_dotfiles() {
 	local ret=0
-	for f in "$pkgdir"/.*; do
+	for f in $(find $pkgdir -name ".*"); do
 		[[ ${f##*/} == . || ${f##*/} == .. ]] && continue
 		error "$(gettext "Dotfile found in package root '%s'")" "$f"
 		ret=1
