--- a/src/gfxstream/guest/platform/linux/LinuxVirtGpuBlob.cpp
+++ b/src/gfxstream/guest/platform/linux/LinuxVirtGpuBlob.cpp.new
@@ -97,10 +97,10 @@
     }
 
     uint8_t* ptr = static_cast<uint8_t*>(
-        mmap64(nullptr, mSize, PROT_WRITE | PROT_READ, MAP_SHARED, mDeviceHandle, map.offset));
+        mmap(nullptr, mSize, PROT_WRITE | PROT_READ, MAP_SHARED, mDeviceHandle, map.offset));
 
     if (ptr == MAP_FAILED) {
-        mesa_loge("mmap64 failed with (%s)", strerror(errno));
+        mesa_loge("mmap failed with (%s)", strerror(errno));
         return nullptr;
     }
 
