--- libmusicbrainz-5.1.0/CMakeLists.txt
+++ libmusicbrainz-5.1.0.new/CMakeLists.txt
--- libmusicbrainz-5.1.0/CMakeLists.txt
+++ libmusicbrainz-5.1.0.new/CMakeLists.txt
@@ -40,12 +40,6 @@
 ADD_SUBDIRECTORY(tests)
 ADD_SUBDIRECTORY(examples)
 
-ADD_CUSTOM_TARGET(docs
-	doxygen
-)
-
-ADD_DEPENDENCIES(docs src_gen)
-
 ADD_CUSTOM_TARGET(dist
     rm -rf "libmusicbrainz-${PROJECT_VERSION}"
     COMMAND git archive --prefix="libmusicbrainz-${PROJECT_VERSION}/" master | gzip -9 > libmusicbrainz-${PROJECT_VERSION}.tar.gz

