diff --git a/src/declarativeimports/core/windowthumbnail.cpp b/src/declarativeimports/core/windowthumbnail.cpp
index 5e9f3f1ec..90f1a0a13 100644
--- a/src/declarativeimports/core/windowthumbnail.cpp
+++ b/src/declarativeimports/core/windowthumbnail.cpp
@@ -6,7 +6,9 @@
 #include "windowthumbnail.h"
 // KF5
 #include <KWindowSystem>
+#if HAVE_X11
 #include <KX11Extras>
+#endif
 // Qt
 #include <QGuiApplication>
 #include <QIcon>
@@ -282,10 +284,12 @@ void WindowThumbnail::setWinId(uint32_t winId)
     if (m_winId == winId) {
         return;
     }
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11() && !KX11Extras::self()->hasWId(winId)) {
         // invalid Id, don't updated
         return;
     }
+#endif
     if (window() && winId == window()->winId()) {
         // don't redirect to yourself
         return;
@@ -395,12 +399,16 @@ bool WindowThumbnail::nativeEventFilter(const QByteArray &eventType, void *messa
 void WindowThumbnail::iconToTexture(WindowTextureProvider *textureProvider)
 {
     QIcon icon;
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11() && KX11Extras::self()->hasWId(m_winId)) {
         icon = KX11Extras::self()->icon(m_winId, boundingRect().width(), boundingRect().height());
     } else {
+#endif
         // fallback to plasma icon
         icon = QIcon::fromTheme(QStringLiteral("plasma"));
+#if HAVE_X11
     }
+#endif
     QImage image = icon.pixmap(boundingRect().size().toSize(), window()->devicePixelRatio()).toImage();
     textureProvider->setTexture(window()->createTextureFromImage(image, QQuickWindow::TextureCanUseAtlas));
 }
diff --git a/src/plasma/private/blureffectwatcher.cpp b/src/plasma/private/blureffectwatcher.cpp
index 6f9694dc6..92c1bb120 100644
--- a/src/plasma/private/blureffectwatcher.cpp
+++ b/src/plasma/private/blureffectwatcher.cpp
@@ -88,13 +88,12 @@ void BlurEffectWatcher::init()
     }
 }
 
-#if HAVE_X11
 bool BlurEffectWatcher::nativeEventFilter(const QByteArray &eventType, void *message, qintptr *result)
 {
     if (KWindowSystem::isPlatformWayland()) {
         return false;
     }
-
+#if HAVE_X11
     Q_UNUSED(result);
     // A faster comparison than eventType != "xcb_generic_event_t"
     // given that eventType can only have the following values:
@@ -117,10 +116,9 @@ bool BlurEffectWatcher::nativeEventFilter(const QByteArray &eventType, void *mes
             Q_EMIT effectChanged(m_effectActive);
         }
     }
-
+#endif
     return false;
 }
-#endif
 
 bool BlurEffectWatcher::isEffectActive() const
 {
@@ -129,10 +127,10 @@ bool BlurEffectWatcher::isEffectActive() const
 
 bool BlurEffectWatcher::fetchEffectActive() const
 {
+#if HAVE_X11
     if (KWindowSystem::isPlatformWayland()) {
         return m_blurManager->isActive();
     }
-
     if (m_property == XCB_ATOM_NONE || !m_x11Interface) {
         return false;
     }
@@ -149,6 +147,9 @@ bool BlurEffectWatcher::fetchEffectActive() const
         }
     }
     return false;
+#else
+    return m_blurManager->isActive();
+#endif
 }
 
 } // namespace Plasma
diff --git a/src/plasma/private/blureffectwatcher_p.h b/src/plasma/private/blureffectwatcher_p.h
index 36f0be511..f7a8c1e8d 100644
--- a/src/plasma/private/blureffectwatcher_p.h
+++ b/src/plasma/private/blureffectwatcher_p.h
@@ -12,7 +12,11 @@
 
 #include <QAbstractNativeEventFilter>
 
+#include <config-plasma.h>
+
+#if HAVE_X11
 #include <xcb/xcb.h>
+#endif
 
 #include <config-plasma.h>
 
@@ -30,9 +34,7 @@ public:
     bool isEffectActive() const;
 
 protected:
-#if HAVE_X11
     bool nativeEventFilter(const QByteArray &eventType, void *message, qintptr *) override;
-#endif
 
 Q_SIGNALS:
     void effectChanged(bool on);
diff --git a/src/plasma/private/theme_p.cpp b/src/plasma/private/theme_p.cpp
index f4917a945..ca4c366ec 100644
--- a/src/plasma/private/theme_p.cpp
+++ b/src/plasma/private/theme_p.cpp
@@ -17,7 +17,9 @@
 #include <KSharedConfig>
 #include <KWindowEffects>
 #include <KWindowSystem>
+#if HAVE_X11
 #include <KX11Extras>
+#endif
 
 namespace Plasma
 {
@@ -96,9 +98,11 @@ ThemePrivate::ThemePrivate(QObject *parent)
     , apiMinor(0)
     , apiRevision(0)
 {
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         compositingActive = KX11Extras::self()->compositingActive();
     }
+#endif
 
     kSvgImageSet = std::unique_ptr<KSvg::ImageSet>(new KSvg::ImageSet);
     kSvgImageSet->setBasePath(QStringLiteral(PLASMA_RELATIVE_DATA_INSTALL_DIR "/desktoptheme/"));
@@ -139,9 +143,11 @@ ThemePrivate::ThemePrivate(QObject *parent)
         scheduleThemeChangeNotification();
     });
 
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         connect(KX11Extras::self(), &KX11Extras::compositingChanged, selectorsUpdateTimer, qOverload<>(&QTimer::start));
     }
+#endif
     updateKSvgSelectors();
 }
 
diff --git a/src/plasmaquick/appletpopup.cpp b/src/plasmaquick/appletpopup.cpp
index 5774c1888..910895390 100644
--- a/src/plasmaquick/appletpopup.cpp
+++ b/src/plasmaquick/appletpopup.cpp
@@ -13,7 +13,9 @@
 
 #include <KConfigGroup>
 #include <KWindowSystem>
+#if HAVE_X11
 #include <KX11Extras>
+#endif
 
 #include "applet.h"
 #include "appletquickitem.h"
@@ -64,12 +66,16 @@ AppletPopup::AppletPopup()
     setAnimated(true);
     setFlags(flags() | Qt::Dialog);
 
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         KX11Extras::setType(winId(), NET::AppletPopup);
     } else {
+#endif
         PlasmaShellWaylandIntegration::get(this)->setRole(QtWayland::org_kde_plasma_surface::role::role_appletpopup);
         PlasmaShellWaylandIntegration::get(this)->setTakesFocus(true);
+#if HAVE_X11
     }
+#endif
 
     auto edgeForwarder = new EdgeEventForwarder(this);
     edgeForwarder->setMargins(padding());
diff --git a/src/plasmaquick/appletpopup.h b/src/plasmaquick/appletpopup.h
index 29982d077..dbcda2535 100644
--- a/src/plasmaquick/appletpopup.h
+++ b/src/plasmaquick/appletpopup.h
@@ -10,6 +10,8 @@
 
 #include <QQuickItem>
 
+#include <config-plasma.h>
+
 #include <plasmaquick/plasmaquick_export.h>
 
 namespace PlasmaQuick
diff --git a/src/plasmaquick/dialog.cpp b/src/plasmaquick/dialog.cpp
index 096c9c2c8..00c46addb 100644
--- a/src/plasmaquick/dialog.cpp
+++ b/src/plasmaquick/dialog.cpp
@@ -24,9 +24,13 @@
 #include <QQuickItem>
 #include <QScreen>
 
+#if HAVE_X11
 #include <KWindowInfo>
+#endif
 #include <KWindowSystem>
+#if HAVE_X11
 #include <KX11Extras>
+#endif
 
 #include <KWindowEffects>
 #include <Plasma/Corona>
@@ -253,15 +257,19 @@ void DialogPrivate::updateTheme()
                                                  theme.backgroundSaturation(),
                                                  mask);
 
+#if HAVE_X11
         if (!KWindowSystem::isPlatformX11() || KX11Extras::compositingActive()) {
+#endif
             if (hasMask) {
                 hasMask = false;
                 q->setMask(QRegion());
             }
+#if HAVE_X11
         } else {
             hasMask = true;
             q->setMask(dialogBackground->mask());
         }
+#endif
         if (q->isVisible()) {
             DialogShadows::instance()->addWindow(q, dialogBackground->enabledBorders());
         }
@@ -710,9 +718,9 @@ QPointF DialogPrivate::positionAdjustedForMainItem(const QPointF &point) const
 
 void DialogPrivate::applyType()
 {
+#if HAVE_X11
     /*QXcbWindowFunctions::WmWindowType*/ int wmType = 0;
 
-#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         switch (type) {
         case Dialog::Normal:
@@ -745,11 +753,11 @@ void DialogPrivate::applyType()
             q->setProperty("_q_xcb_wm_window_type", wmType);
         }
     }
-#endif
 
     if (!wmType && type != Dialog::Normal && KWindowSystem::isPlatformX11()) {
         KX11Extras::setType(q->winId(), static_cast<NET::WindowType>(type));
     }
+#endif
     if (q->flags() & Qt::WindowStaysOnTopHint) {
         // If the AppletPopup type is not explicitly requested, then use the Dock type in this case
         // to avoid bug #454635.
@@ -812,6 +820,7 @@ void DialogPrivate::applyType()
         }
     }
 
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         if (type == Dialog::Dock || type == Dialog::Notification || type == Dialog::OnScreenDisplay || type == Dialog::CriticalNotification) {
             KX11Extras::setOnAllDesktops(q->winId(), true);
@@ -819,6 +828,7 @@ void DialogPrivate::applyType()
             KX11Extras::setOnAllDesktops(q->winId(), false);
         }
     }
+#endif
 
     PlasmaShellWaylandIntegration::get(q)->setTakesFocus(!q->flags().testFlag(Qt::WindowDoesNotAcceptFocus));
 }
@@ -1080,11 +1090,13 @@ QPoint Dialog::popupPosition(QQuickItem *item, const QSize &size)
     // if the item is in a window that ignores WM we want to position the popups outside
     bool outsideParentWindow = (item->window()->flags() & Qt::X11BypassWindowManagerHint) && item->window()->mask().isNull();
 
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         // on X11 we also consider windows with the type Dock
         const KWindowInfo winInfo(item->window()->winId(), NET::WMWindowType);
         outsideParentWindow = outsideParentWindow || (winInfo.windowType(NET::AllTypesMask) == NET::Dock && item->window()->mask().isNull());
     }
+#endif
 
     QRect parentGeometryBounds;
     if (outsideParentWindow) {
@@ -1358,9 +1370,11 @@ void Dialog::showEvent(QShowEvent *event)
         DialogShadows::instance()->addWindow(this, d->dialogBackground->enabledBorders());
     }
 
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         KX11Extras::setState(winId(), NET::SkipTaskbar | NET::SkipPager | NET::SkipSwitcher);
     }
+#endif
     QQuickWindow::showEvent(event);
 }
 
diff --git a/src/plasmaquick/plasmawindow.cpp b/src/plasmaquick/plasmawindow.cpp
index 4aaf54821..d944695ed 100644
--- a/src/plasmaquick/plasmawindow.cpp
+++ b/src/plasmaquick/plasmawindow.cpp
@@ -11,9 +11,13 @@
 #include <QMarginsF>
 #include <QQuickItem>
 
+#include <config-plasma.h>
+
 #include <KWindowEffects>
 #include <KWindowSystem>
+#if HAVE_X11
 #include <KX11Extras>
+#endif
 
 #include <Plasma/Theme>
 
@@ -143,9 +147,11 @@ void PlasmaWindow::showEvent(QShowEvent *e)
 {
     // EWMH states that the state is reset every hide
     // Qt supports external factors setting state before the next show
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         KX11Extras::setState(winId(), NET::SkipTaskbar | NET::SkipPager | NET::SkipSwitcher);
     }
+#endif
     QQuickWindow::showEvent(e);
 }
 
@@ -171,11 +177,15 @@ void PlasmaWindowPrivate::handleFrameChanged()
                                              theme.backgroundSaturation(),
                                              mask);
 
+#if HAVE_X11
     if (!KWindowSystem::isPlatformX11() || KX11Extras::compositingActive()) {
+#endif
         q->setMask(QRegion());
+#if HAVE_X11
     } else {
         q->setMask(mask);
     }
+#endif
 }
 
 void PlasmaWindowPrivate::updateMainItemGeometry()
