--- libplasma-6.4.4/src/declarativeimports/core/windowthumbnail.cpp
+++ libplasma-6.4.4-new/src/declarativeimports/core/windowthumbnail.cpp
@@ -6,7 +6,9 @@
 #include "windowthumbnail.h"
 // KF5
 #include <KWindowSystem>
+#if HAVE_X11
 #include <KX11Extras>
+#endif
 // Qt
 #include <QGuiApplication>
 #include <QIcon>
@@ -280,10 +282,12 @@
     if (m_winId == winId) {
         return;
     }
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11() && !KX11Extras::self()->hasWId(winId)) {
         // invalid Id, don't updated
         return;
     }
+#endif
     if (window() && winId == window()->winId()) {
         // don't redirect to yourself
         return;
@@ -393,12 +397,16 @@
 void WindowThumbnail::iconToTexture(WindowTextureProvider *textureProvider)
 {
     QIcon icon;
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11() && KX11Extras::self()->hasWId(m_winId)) {
         icon = KX11Extras::self()->icon(m_winId, boundingRect().width(), boundingRect().height());
     } else {
+#endif
         // fallback to plasma icon
         icon = QIcon::fromTheme(QStringLiteral("plasma"));
+#if HAVE_X11
     }
+#endif
     QImage image = icon.pixmap(boundingRect().size().toSize(), window()->devicePixelRatio()).toImage();
     textureProvider->setTexture(window()->createTextureFromImage(image, QQuickWindow::TextureCanUseAtlas));
 }
--- libplasma-6.4.4/src/plasma/private/contrasteffectwatcher.cpp
+++ libplasma-6.4.4-new/src/plasma/private/contrasteffectwatcher.cpp
@@ -88,13 +88,12 @@
     }
 }
 
-#if HAVE_X11
 bool ContrastEffectWatcher::nativeEventFilter(const QByteArray &eventType, void *message, qintptr *result)
 {
     if (KWindowSystem::isPlatformWayland()) {
         return false;
     }
-
+#if HAVE_X11
     Q_UNUSED(result);
     // A faster comparison than eventType != "xcb_generic_event_t"
     // given that eventType can only have the following values:
@@ -118,9 +117,9 @@
         }
     }
 
+#endif
     return false;
 }
-#endif
 
 bool ContrastEffectWatcher::isEffectActive() const
 {
@@ -132,7 +131,7 @@
     if (KWindowSystem::isPlatformWayland()) {
         return m_contrastManager->isActive();
     }
-
+#if HAVE_X11
     if (m_property == XCB_ATOM_NONE || !m_x11Interface) {
         return false;
     }
@@ -148,6 +147,7 @@
             return true;
         }
     }
+#endif
     return false;
 }
 
--- libplasma-6.4.4/src/plasma/private/contrasteffectwatcher_p.h
+++ libplasma-6.4.4-new/src/plasma/private/contrasteffectwatcher_p.h
@@ -12,10 +12,12 @@
 
 #include <QAbstractNativeEventFilter>
 
-#include <xcb/xcb.h>
-
 #include <config-plasma.h>
 
+#if HAVE_X11
+#include <xcb/xcb.h>
+#endif
+
 namespace Plasma
 {
 class ContrastManager;
@@ -30,9 +32,7 @@
     bool isEffectActive() const;
 
 protected:
-#if HAVE_X11
     bool nativeEventFilter(const QByteArray &eventType, void *message, qintptr *) override;
-#endif
 
 Q_SIGNALS:
     void effectChanged(bool on);
--- libplasma-6.4.4/src/plasma/private/theme_p.cpp
+++ libplasma-6.4.4-new/src/plasma/private/theme_p.cpp
@@ -17,7 +17,9 @@
 #include <KSharedConfig>
 #include <KWindowEffects>
 #include <KWindowSystem>
+#if HAVE_X11
 #include <KX11Extras>
+#endif
 
 namespace Plasma
 {
@@ -96,9 +98,11 @@
     , apiMinor(0)
     , apiRevision(0)
 {
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         compositingActive = KX11Extras::self()->compositingActive();
     }
+#endif
 
     kSvgImageSet = std::unique_ptr<KSvg::ImageSet>(new KSvg::ImageSet);
     kSvgImageSet->setBasePath(QStringLiteral(PLASMA_RELATIVE_DATA_INSTALL_DIR "/desktoptheme/"));
@@ -139,9 +143,11 @@
         scheduleThemeChangeNotification();
     });
 
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         connect(KX11Extras::self(), &KX11Extras::compositingChanged, selectorsUpdateTimer, qOverload<>(&QTimer::start));
     }
+#endif
     updateKSvgSelectors();
 }
 
--- libplasma-6.4.4/src/plasmaquick/appletpopup.cpp
+++ libplasma-6.4.4-new/src/plasmaquick/appletpopup.cpp
@@ -12,7 +12,9 @@
 
 #include <KConfigGroup>
 #include <KWindowSystem>
+#if HAVE_X11
 #include <KX11Extras>
+#endif
 #include <QSize>
 
 #include "applet.h"
@@ -65,11 +67,15 @@
     setAnimated(true);
     setFlags(flags() | Qt::Dialog);
 
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         KX11Extras::setType(winId(), NET::AppletPopup);
     } else {
+#endif
         PlasmaShellWaylandIntegration::get(this)->setRole(QtWayland::org_kde_plasma_surface::role::role_appletpopup);
+#if HAVE_X11
     }
+#endif
 
     auto edgeForwarder = new EdgeEventForwarder(this);
     edgeForwarder->setMargins(padding());
--- libplasma-6.4.4/src/plasmaquick/appletpopup.h
+++ libplasma-6.4.4-new/src/plasmaquick/appletpopup.h
@@ -10,6 +10,8 @@
 
 #include <QQuickItem>
 
+#include <config-plasma.h>
+
 #include <plasmaquick/plasmaquick_export.h>
 
 namespace PlasmaQuick
--- libplasma-6.4.4/src/plasmaquick/dialog.cpp
+++ libplasma-6.4.4-new/src/plasmaquick/dialog.cpp
@@ -24,9 +24,13 @@
 #include <QQuickItem>
 #include <QScreen>
 
+#if HAVE_X11
 #include <KWindowInfo>
+#endif
 #include <KWindowSystem>
+#if HAVE_X11
 #include <KX11Extras>
+#endif
 
 #include <KWindowEffects>
 #include <Plasma/Corona>
@@ -253,15 +257,19 @@
                                                  theme.backgroundSaturation(),
                                                  mask);
 
+#if HAVE_X11
         if (!KWindowSystem::isPlatformX11() || KX11Extras::compositingActive()) {
+#endif
             if (hasMask) {
                 hasMask = false;
                 q->setMask(QRegion());
             }
+#if HAVE_X11
         } else {
             hasMask = true;
             q->setMask(dialogBackground->mask());
         }
+#endif
         if (q->isVisible()) {
             DialogShadows::instance()->addWindow(q, dialogBackground->enabledBorders());
         }
@@ -710,9 +718,9 @@
 
 void DialogPrivate::applyType()
 {
+#if HAVE_X11
     /*QXcbWindowFunctions::WmWindowType*/ int wmType = 0;
 
-#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         switch (type) {
         case Dialog::Normal:
@@ -745,11 +753,11 @@
             q->setProperty("_q_xcb_wm_window_type", wmType);
         }
     }
-#endif
 
     if (!wmType && type != Dialog::Normal && KWindowSystem::isPlatformX11()) {
         KX11Extras::setType(q->winId(), static_cast<NET::WindowType>(type));
     }
+#endif
     if (q->flags() & Qt::WindowStaysOnTopHint) {
         // If the AppletPopup type is not explicitly requested, then use the Dock type in this case
         // to avoid bug #454635.
@@ -812,6 +820,7 @@
         }
     }
 
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         if (type == Dialog::Dock || type == Dialog::Notification || type == Dialog::OnScreenDisplay || type == Dialog::CriticalNotification) {
             KX11Extras::setOnAllDesktops(q->winId(), true);
@@ -819,6 +828,7 @@
             KX11Extras::setOnAllDesktops(q->winId(), false);
         }
     }
+#endif
 
     PlasmaShellWaylandIntegration::get(q)->setTakesFocus(!q->flags().testFlag(Qt::WindowDoesNotAcceptFocus));
 }
@@ -1080,11 +1090,13 @@
     // if the item is in a window that ignores WM we want to position the popups outside
     bool outsideParentWindow = (item->window()->flags() & Qt::X11BypassWindowManagerHint) && item->window()->mask().isNull();
 
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         // on X11 we also consider windows with the type Dock
         const KWindowInfo winInfo(item->window()->winId(), NET::WMWindowType);
         outsideParentWindow = outsideParentWindow || (winInfo.windowType(NET::AllTypesMask) == NET::Dock && item->window()->mask().isNull());
     }
+#endif
 
     QRect parentGeometryBounds;
     if (outsideParentWindow) {
@@ -1358,9 +1370,11 @@
         DialogShadows::instance()->addWindow(this, d->dialogBackground->enabledBorders());
     }
 
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         KX11Extras::setState(winId(), NET::SkipTaskbar | NET::SkipPager | NET::SkipSwitcher);
     }
+#endif
     QQuickWindow::showEvent(event);
 }
 
--- libplasma-6.4.4/src/plasmaquick/plasmawindow.cpp
+++ libplasma-6.4.4-new/src/plasmaquick/plasmawindow.cpp
@@ -11,9 +11,13 @@
 #include <QMarginsF>
 #include <QQuickItem>
 
+#include <config-plasma.h>
+
 #include <KWindowEffects>
 #include <KWindowSystem>
+#if HAVE_X11
 #include <KX11Extras>
+#endif
 
 #include <Plasma/Theme>
 
@@ -143,9 +147,11 @@
 {
     // EWMH states that the state is reset every hide
     // Qt supports external factors setting state before the next show
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         KX11Extras::setState(winId(), NET::SkipTaskbar | NET::SkipPager | NET::SkipSwitcher);
     }
+#endif
     QQuickWindow::showEvent(e);
 }
 
@@ -171,11 +177,15 @@
                                              theme.backgroundSaturation(),
                                              mask);
 
+#if HAVE_X11
     if (!KWindowSystem::isPlatformX11() || KX11Extras::compositingActive()) {
+#endif
         q->setMask(QRegion());
+#if HAVE_X11
     } else {
         q->setMask(mask);
     }
+#endif
 }
 
 void PlasmaWindowPrivate::updateMainItemGeometry()
