--- okular-25.12.3.orig/CMakeLists.txt
+++ okular-25.12.3/CMakeLists.txt
@@ -206,7 +206,7 @@
     else()
         find_package(Qt6Gui REQUIRED COMPONENTS Private)
     endif()
-    set(HAVE_X11 TRUE)
+    set(HAVE_X11 FALSE)
 else()
     set(HAVE_X11 FALSE)
 endif()
--- okular-25.12.3.orig/shell/shell.cpp
+++ okular-25.12.3/shell/shell.cpp
@@ -28,10 +28,6 @@
 #include <KRecentFilesAction>
 #include <KSharedConfig>
 #include <KStandardAction>
-#if !defined(Q_OS_WIN) && !defined(Q_OS_OSX) && !defined(Q_OS_HAIKU)
-#include <KStartupInfo>
-#include <KWindowInfo>
-#endif
 #include <KToggleFullScreenAction>
 #include <KToolBar>
 #include <KUrlMimeData>
@@ -448,13 +444,6 @@
         return false;
     }
 
-#if !defined(Q_OS_WIN) && !defined(Q_OS_OSX) && !defined(Q_OS_HAIKU)
-    const KWindowInfo winfo(window()->effectiveWinId(), NET::WMDesktop);
-    if (winfo.desktop() != desktop) {
-        return false;
-    }
-#endif
-
     return true;
 }
 
@@ -720,16 +709,6 @@
 
 void Shell::tryRaise(const QString &startupId)
 {
-#if !defined(Q_OS_WIN) && !defined(Q_OS_OSX) && !defined(Q_OS_HAIKU)
-    if (KWindowSystem::isPlatformWayland()) {
-        KWindowSystem::setCurrentXdgActivationToken(startupId);
-    } else if (KWindowSystem::isPlatformX11()) {
-        KStartupInfo::setNewStartupId(window()->windowHandle(), startupId.toUtf8());
-    }
-#else
-    Q_UNUSED(startupId);
-#endif
-
     KWindowSystem::activateWindow(window()->windowHandle());
 }
 
