diff --git a/main/Makefile b/main/Makefile
index 1da18a8..8bb348e 100644
--- a/main/Makefile
+++ b/main/Makefile
@@ -150,7 +150,6 @@ install:
 	do \
 	   $(INSTALL) $(BINDIR)/flite_$$i $(DESTDIR)$(INSTALLBINDIR); \
 	done
-	$(INSTALL) -m 755 $(BINDIR)/flite_time $(DESTDIR)$(INSTALLBINDIR)
 #       The libraries: static and shared (if built)
 	cp -pd $(flite_LIBS_deps) $(DESTDIR)$(INSTALLLIBDIR)
 ifdef SHFLAGS
