--- sphinxcontrib-jsmath-1.0.1/sphinxcontrib/jsmath/__init__.py
+++ sphinxcontrib-jsmath-1.0.1/sphinxcontrib/jsmath/__init__.py
@@ -15,8 +15,6 @@
 from docutils import nodes
 from sphinx.application import Sphinx
 from sphinx.builders.html import StandaloneHTMLBuilder
-from sphinx.domains.math import MathDomain
-from sphinx.environment import BuildEnvironment
 from sphinx.errors import ExtensionError
 from sphinx.locale import get_translation
 from sphinx.util.math import get_node_equation_number
@@ -62,7 +60,13 @@
     raise nodes.SkipNode
 
 
-def install_jsmath(app: Sphinx, env: BuildEnvironment) -> None:
+def install_jsmath(
+    app: Sphinx,
+    pagename: str,
+    templatename: str,
+    context: dict[str, Any],
+    event_arg: Any,
+) -> None:
     if app.builder.format != 'html' or app.builder.math_renderer_name != 'jsmath':  # type: ignore  # NOQA
         return
     if not app.config.jsmath_path:
@@ -70,8 +74,8 @@
                              'jsmath extension to work')
 
     builder = cast(StandaloneHTMLBuilder, app.builder)
-    domain = cast(MathDomain, env.get_domain('math'))
-    if domain.has_equations():
+    page_has_equations = context.get('has_maths_elements', True)
+    if app.registry.html_assets_policy == 'always' or page_has_equations:
         # Enable jsmath only if equations exists
         builder.add_js_file(app.config.jsmath_path)
 
@@ -84,7 +88,7 @@
                                (html_visit_displaymath, None))
 
     app.add_config_value('jsmath_path', '', False)
-    app.connect('env-updated', install_jsmath)
+    app.connect('html-page-context', install_jsmath)
     return {
         'version': __version__,
         'parallel_read_safe': True,
--- sphinxcontrib-jsmath-1.0.1/tests/test_jsmath.py
+++ sphinxcontrib-jsmath-1.0.1/tests/test_jsmath.py
@@ -9,6 +9,7 @@
 """
 
 import pytest
+import sphinx
 
 
 @pytest.mark.sphinx('html', testroot='basic')
@@ -16,6 +17,7 @@
     app.builder.build_all()
     content = (app.outdir / 'math.html').read_text()
     print(content)
+    assert 'jsmath.js' in content
     assert '<div class="math notranslate nohighlight">\nE = mc^2</div>' in content
     assert ('<span class="eqno">(1)<a class="headerlink" href="#equation-pythagorean" '
             'title="Permalink to this equation">¶</a></span>'
@@ -35,6 +37,7 @@
     app.builder.build_all()
 
     content = (app.outdir / 'math.html').read_text()
+    assert 'jsmath.js' in content
     assert '<div class="math notranslate nohighlight">\nE = mc^2</div>' in content
     assert ('<span class="eqno">(1.1)<a class="headerlink" href="#equation-pythagorean" '
             'title="Permalink to this equation">¶</a></span>'
@@ -48,6 +51,10 @@
     assert '<a class="reference internal" href="#equation-pythagorean">(1.1)</a>' in content
 
 
+@pytest.mark.skipif(
+    sphinx.version_info < (8, 2),
+    reason='Sphinx < 8.2 does not have `has_maths_elements` in context',
+)
 @pytest.mark.sphinx('html', testroot='nomath')
 def test_disabled_when_equations_not_found(app, status, warning):
     app.builder.build_all()
